common/thorhelper/thorxmlwrite.cpp

```diff
@@ -685,6 +685,590 @@ StringBuffer &buildJsonFooter(StringBuffer  &footer, const char *suppliedFooter,
+        return;
+
+    MemoryAttr tempBuffer;
+    char * temp;
+    if (len <= 100)
+        temp = (char *)alloca(len);
+    else
+        temp = (char *)tempBuffer.allocate(len);
+    rtlQStrToStr(len, temp, len, field);
+    addStringField(len, temp, fieldName);
+}
+
+void CommonCSVWriter::outputUtf8(unsigned len, const char* field, const char* fieldName)
+{
+    if (!checkHeaderName(fieldName))
+        return;
+
+    addStringField(rtlUtf8Size(len, field), field, fieldName);
+}
+
+void CommonCSVWriter::outputNumericString(const char* field, const char* fieldName)
+{
+    if (!checkHeaderName(fieldName))
+        return;
+
+    addStringField((size32_t)strlen(field), field, fieldName);
+}
+
+void CommonCSVWriter::outputBeginNested(const char* fieldName, bool simpleNested)
+{
+    if (!fieldName || !*fieldName)
+        return;
+
+    if (readingCSVHeader)
+    {
+        addCSVHeader(fieldName, NULL, true, simpleNested);
+        if (simpleNested) //ECL SET has only one column (parent name should be used as column name).
+            columnID++;
+        nestedHeaderLayerID++;
+        currentParentXPath.append(fieldName).append("/");
+        return;
+    }
+
+    if (!strieq(fieldName, "row"))
+    {//A nested item begins.
+        nestedContentLayerID++;
+        currentParentXPath.append(fieldName).append("/");
+    }
+    else if (!currentParentXPath.isEmpty())
+    {//A new row begins inside a nested item.
+        CCSVItem* item = getParentCSVItem();
+        if (!item)
+            return;
+
+        unsigned rowCount = item->getRowCount();
+        if (rowCount > 0)
+        {//Starting from the second result row, the NextRowIDs of every children are reset based on the last result row.
+            StringBuffer path = currentParentXPath;
+            path.setLength(path.length() - 1);
+            setChildrenNextRowID(path.str(), getMaxChildrenNextRowID(path.str()));
+        }
+
+        item->setCurrentRowEmpty(true);
+    }
+}
+
+void CommonCSVWriter::outputEndNested(const char* fieldName)
+{
+    if (!fieldName || !*fieldName)
+        return;
+
+    if (readingCSVHeader)
+    {
+        removeFieldFromCurrentParentXPath(fieldName);
+        nestedHeaderLayerID--;
+        return;
+    }
+
+    if (!strieq(fieldName, "row"))
+    {//This is an end of a nested item.
+        removeFieldFromCurrentParentXPath(fieldName);
+        nestedContentLayerID--;
+    }
+    else
+    {//A row ends inside the nested item
+        if (!currentParentXPath.isEmpty())
+        {
+            CCSVItem* item = getParentCSVItem();
+            if (item && !item->getCurrentRowEmpty())
+            {
+                item->incrementRowCount();
+                item->setCurrentRowEmpty(true);
+            }
+        }
+        if (nestedContentLayerID == 0) //outputEndNested Row at layer 0 indicates the end of a WU Result row.
+            endRecord();
+    }
+}
+
+IXmlWriterExt& CommonCSVWriter::clear()
+{
+    out.clear();
```

Comment:
There are many more variables in this class... do any of them need to be cleared as well?